File: src/system/MacAddress.java
/*
 * Copyright (c) 2015. This code is a LogosProg property. All Rights Reserved.
 */

package system;

import java.net.InetAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.net.UnknownHostException;

/**
 * Created by forando on 14.10.15.
 */
public class MacAddress {

    public static String getMacAddress(){
        String macAdd = null;

        InetAddress ip;
        try {

            ip = InetAddress.getLocalHost();
            System.out.println("Current IP address : " + ip.getHostAddress());

            NetworkInterface network = NetworkInterface.getByInetAddress(ip);

            byte[] mac = network.getHardwareAddress();

            System.out.print("Current MAC address : ");

            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < mac.length; i++) {
                sb.append(String.format("%02X%s", mac[i], (i < mac.length - 1) ? "-" : ""));
            }
            macAdd = sb.toString();

        } catch (UnknownHostException e) {

            e.printStackTrace();

        } catch (SocketException e){

            e.printStackTrace();

        }

        return macAdd;
    }
}

File: src/main/kotlin/com/logosprog/mainutils/system/MacAddress.kt
/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

/*
 * Copyright (c) 2016. This code is a LogosProg property. All Rights Reserved.
 */

@file:Suppress("unused")

package com.logosprog.mainutils.system

import java.net.InetAddress
import java.net.NetworkInterface

/**
 * Gets a mac address of the machine
 */

fun macAddress(): String{
    val ip: InetAddress = InetAddress.getByName("127.0.0.1")

    val network = NetworkInterface.getByInetAddress(ip)

    val mac: ByteArray = network.hardwareAddress

    val sb = StringBuilder()

    for ((i, byte) in mac.withIndex()){
        sb.append("$byte${if(i < mac.size - 1) "-" else ""}")
    }

    return sb.toString()
}